### PR TITLE
Behroozi 2019 SMHM data (fit for all z)

### PR DESCRIPTION
**Pull request by EvgeniiChaikin**
- Behroozi 2019 SMHM fitting function (for all z) has been added to VR
- Redshifts will now show up in obs-data curves' labels

Example of the new stellar mass halo mass plot:

stellar_mass_halo_mass_centrals_30

### CODE REVIEW

**Comment by MatthieuSchaller** [comment]
At z>0, is there any point in showing the z=0 data? 

Also, we should be careful with M_vir -> M_200. Maybe we really need to have separate plots here. One plot with M_200_crit and Moster and one with M_vir + Behroozi data.

**Comment by EvgeniiChaikin** [comment]
@MatthieuSchaller I wrote a conversion script that converts raw Behroozi 2019 data into `Behroozi2019.hdf5`. In that script, for now I simply divide `M_vir` by `1.2` to get `M_200`, and `M_200` is then plotted. I will add this script to the Observational-data repository when I get access to push commits.

The correction factor `1.2` is what you recently calculated based on the M_200 vs M_vir data in EAGLE. The definition of M halo in Behroozi  2013 and 2019 papers is the same (if we assume `M_{vir, peak} \approx M_{vir}`), that is why using the same correction factor for the 2019 data is justified.

I think if we have no high-z data, then plotting z=0 data is better than plotting no observational data.

**Comment by MatthieuSchaller** [comment]
The 1.2 is only valid at z=0.

**Comment by EvgeniiChaikin** [comment]
Shall we then make the same comparison based on EAGLE data but for `z>0`?

**Comment by MatthieuSchaller** [comment]
We could, but why not do this correctly and make two plots? 

The 1.2 correction is not good anyway if we are trying to do things well. It is a function of mass and cosmology plus it ignores the scatter. Since we have all we need to do things correctly, I don't see why we don't do two plots.

**Comment by JBorrow** [comment]
Yes, please leave the data un-corrected and leave them tagged as what they are (M_200,crit v.s. M_vir). We should then create two plots if we actually want to see them both - we have enough of these M_*/M_H lines anyway...

**Comment by EvgeniiChaikin** [comment]
@JBorrow The new Berhoozi fitting function in this PR is fine, i.e. it does not do any internal conversion from M_200 to M_crit or anything like that. If there were any conversion of data, it would be in a new Berhoozi script which is part of the observational-data repository.

Btw, In this PR, the new fitting function is called `behroozi_2019_raw`. In the same file, there is also `behroozi_raw` which refers to Behroozi 2013 et al. Should we change `behroozi_raw - > behroozi_2013_raw`?

**Comment by JBorrow** [comment]
Ah okay, I now see what the PR does.

Please do not submit more than one feature change within a given pull request.

File: velociraptor/fitting_formulae/smhmr.py [hunk 1]
```diff
@@ -164,3 +164,92 @@ def behroozi_raw(z, Mhalo):
...
+        1.149615e02,
+    ]
+
+    names = (
+        "EFF_0 EFF_0_A EFF_0_A2 EFF_0_Z M_1 M_1_A M_1_A2 M_1_Z ALPHA ALPHA_A "
+        "ALPHA_A2 ALPHA_Z BETA BETA_A BETA_Z DELTA GAMMA GAMMA_A GAMMA_Z CHI2".split(
+            " "
+        )
+    )
+    params = dict(zip(names, param_list))
```
Comment (JBorrow) [inline velociraptor/fitting_formulae/smhmr.py:216]: Why is this not just a dictionary from the start?
```
params = {
  "EFF_0" : -1.4...,
  ...
}
```

File: velociraptor/observations/objects.py [hunk 2]
```diff
@@ -569,13 +569,16 @@ def plot_on_axes(self, axes: Axes, errorbar_kwargs: Union[dict, None] = None):
         elif self.plot_as == "line":
             kwargs["zorder"] = line_zorder
 
+        # Make both the data name and redshift appear in the legend
+        data_label = self.citation + " ($z={:.1f}$)".format(self.redshift)
```
Comment (JBorrow) [inline velociraptor/observations/objects.py:573]: Please avoid "adding" strings to concatenate them. This should be:

`data_label = f"{self.citation} ($z={self.redshift:.1f}$)"`

File: velociraptor/fitting_formulae/smhmr.py (hunk 1 above)
Comment (EvgeniiChaikin) [reply velociraptor/fitting_formulae/smhmr.py]: That's how it was in the original version. I just pushed a commit so now it looks as you suggested.

File: velociraptor/observations/objects.py (hunk 2 above)
Comment (EvgeniiChaikin) [reply velociraptor/observations/objects.py]: Done

**Comment by JBorrow** [comment]
Thanks!

**Pull request merged**